blenderproc/python/writer/GifWriterUtility.py

```diff
@@ -0,0 +1,210 @@
+                for number, frame in enumerate(output_data_dict[key]):
+                    vis_data(key=key, 
+                            data=frame, 
+                            save_to_file=tmp_dir.name + f'/{number}.png')
+        return to_animate
+
+    @staticmethod
+    def _sort_method(item) -> int:
+        """ Order the images due to the file numbering """
+        _, file_name = item.rsplit('/', maxsplit=1)
+        file_number, _ = file_name.split('.')
+        return int(file_number)
+
+    @staticmethod
+    def _look_for_existing_output(output_dir_path: str, 
+                                append_to_existing_output: bool, 
+                                name_ending: str) -> int:
+        """ 
+        Looks for the highest existing #.gif number 
+        and adapts respectivly 
+        """
+        if append_to_existing_output:
+            gif_number = 0
+            for path in os.listdir(output_dir_path):
+                if path.endswith(name_ending):
+                    index = path[:-len(name_ending)]
+                    if index.isdigit():
+                        gif_number = max(gif_number, int(index) + 1)
+        else:
+            gif_number = 0
+        return gif_number
+
+    @staticmethod
+    def _write_to_gif(to_animate: Dict[str, str], 
+                    output_dir_path: str, 
+                    append_to_existing_output: bool, 
+                    time_constant: int, 
+                    time_direction: bool) -> None:
+        """
+        loads all .png files from each specific temporary folder 
+        and concatenates them to a single gif file respectively 
+        """
+        for key, tmp_dir in to_animate.items():
+            print('gif for ' + key + ' at ' + tmp_dir.name)
+            frames = []
+            # Loads all .png images as paths to a list
+            frame_paths = glob.glob(tmp_dir.name + '/*.png')
+            # sorts list in size w.r.t. given key
+            frame_paths.sort(key = GifWriterUtility._sort_method, 
+                            reverse=time_direction)
+            
+            # loads actual picture data as frames
+            for frame_path in frame_paths:
+                new_frame = Image.open(frame_path)
+                frames.append(new_frame)
+
+            name_ending = f"_{key}_animation.gif"
+            gif_number = GifWriterUtility._look_for_existing_output(
+                                            output_dir_path, 
+                                            append_to_existing_output, 
+                                            name_ending)
+            file_name = str(gif_number) + name_ending
+            file = os.path.join(output_dir_path, file_name)
+            frames[0].save(file,
+                            format ='GIF',
+                            append_images = frames[1:],
+                            save_all = True,
+                            duration = time_constant,
+                            loop = 0)
```

Comment:
Does this use the first loaded png image? Why does the final gif contain the `gif_number`?